Interfaces/API/Job.py

```diff
@@ -1005,19 +1010,28 @@ def _toJDL( self, xmlFile = '' ): #messy but need to account for xml file being
     for param in paramList:
       paramsDict[param.getName()] = {'type':param.getType(), 'value':param.getValue()}
 
-    scriptname = 'jobDescription.xml'
     arguments = []
-    if self.script:
-      if os.path.exists( self.script ):
-        scriptname = os.path.abspath( self.script )
-        self.log.verbose( 'Found script name %s' % scriptname )
+    scriptname = 'jobDescription.xml'
+
+    if jobDescriptionObject is None:
+      # if we are here it's because there's a real file, on disk, that is named 'jobDescription.xml'
+      if self.script:
```

Comment:
Well, I didn't change this part, but you are right. I just added a printout (only), because this method seems to me that anyway is not supposed to fail... I guess it's not the best thing anyway, but changing it properly is not for this PR (IMHO there are also methods, in this API, that should be moved, not only adjusted).